Publisher/Sources/DefaultPublisher.swift

```diff
@@ -149,13 +149,25 @@ extension DefaultPublisher: PublisherObjectiveC {
             }
         }
     }
+    
+    @objc
+    func stop(onSuccess: @escaping (() -> Void), onError: @escaping ((ErrorInformation) -> Void)) {
+        self.stop { result in
+            switch result {
+            case .success:
+                onSuccess()
+            case .failure(let error):
+                onError(error)
+            }
+        }
+    }
 }
 
 // MARK: Threading events handling
 extension DefaultPublisher {
     private func enqueue(event: PublisherEvent) {
         logger.trace("Received event: \(event)")
-        performOnWorkingThread { [weak self] in
+        performOnWorkingThread { [weak self] in            
             switch event {
```

Comment:
I'm not sure if you're handling the `stopped` state in all of those events :thinking: For example, I've not found any `stopped` related code in the `performEnhancedLocationChanged()` method. So if I'm right you'll try to send any location updates that stacked up during the stopping procedure after the publisher is stopped. :wink:

Reply:
Updated:
https://github.com/ably/ably-asset-tracking-cocoa/blob/feature/ATS-266-Publisher_stop/Publisher/Sources/DefaultPublisher.swift#L452